services/sources/clacso_service.py

```diff
@@ -0,0 +1,143 @@
+from datetime import datetime, timedelta, timezone
+from io import BytesIO
+from lxml import etree
+from typing import Optional
+import os
+import requests
+
+from constants.get_constants import get_constants
+from logger import create_log
+from mappings.clacso import CLACSOMapping
+from .source_service import SourceService
+from managers import RabbitMQManager, S3Manager, DOABLinkManager
+
+logger = create_log(__name__)
+
+class CLACSOService(SourceService):
```

Comment:
Is the plan to use the DSpaceService once Jackie's PR is merged in?

Reply:
That is the plan but I think including a verification method for the has_part field to check if the field has a pdf or epub link and a verification method for the medium (this is more for CLACSO since DOAB mapping doesn't include the medium). These methods should help to achieve a 100% FRBR and cluster rate for CLACSO records since non monographic material will be filtered out by the medium and only CLACSO records with an available pdf or epub should be ingested as well.

Reply:
I just merged in my PR. I am available to pair if you need help integrating the DSpaceService here.